pkg/controller/handler.go

```diff
@@ -227,6 +228,28 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 				ResetFinalizers(mutatedWf)
 			}
 		}
+
+		// ExecutionNotFound error is returned when flyteadmin is missing the workflow. This is not
+		// a valid state unless we are experiencing a race condition where the workflow has not yet
+		// been inserted into the db (ie. workflow phase is WorkflowPhaseReady).
+		if err != nil && eventsErr.IsNotFound(err) && w.GetExecutionStatus().GetPhase() != v1alpha1.WorkflowPhaseReady {
+			t.Stop()
+			logger.Errorf(ctx, "Failed to process workflow, failing: %s", err)
+
+			// We set the workflow status to failing to abort any active tasks in the next round.
+			mutableW := w.DeepCopy()
+			mutableW.Status.UpdatePhase(v1alpha1.WorkflowPhaseFailing, "Workflow execution is missing in flyteadmin, aborting", &core.ExecutionError{
```

Comment:
why failing? I think we should move to failed right, and also remove the finalizer etc?

Reply:
I was setting to failing to leverage the existing graceful shutdown mechanisms during the transition from failing to failed. For example, propagating Abort throughout the workflow to execute KillTask on any active k8s resources. Is this not as critical as I imagined? It's certainly easier to set to failed and be done with it.

Reply:
https://docs.flyte.org/en/latest/concepts/state_machine.html#divedeep-state-machine